include/swift/SILOptimizer/Analysis/PassManagerVerifierAnalysis.h

```diff
@@ -0,0 +1,65 @@
+//===--- PassManagerVerifierAnalysis.h ------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2018 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_SILOPTIMIZER_ANALYSIS_PASSMANAGERVERIFIERANALYSIS_H
+#define SWIFT_SILOPTIMIZER_ANALYSIS_PASSMANAGERVERIFIERANALYSIS_H
+
+#include "swift/SIL/SILFunction.h"
+#include "swift/SILOptimizer/Analysis/Analysis.h"
+#include "llvm/ADT/DenseSet.h"
+
+namespace swift {
+
+/// An analysis that validates that the pass manager properly sends add/delete
+/// messages as functions are added/deleted from the module.
```

Comment:
I'd like to see this comment explain how it "validates the pass manager". Such as, "It initializes its own set of "live" functions with the SILModule's current state. It continually updates its own set of live functions by tracking the passmanager's function add and delete notifications. When the "verify()" method is called, it reconciles it's own set of live functions with the current state of the SILModule".